### PR TITLE
using tip manager

### PR DESCRIPTION
**Pull request by MithrilMan**
begin of implementation of https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/3241

incomplete, do not merge and discuss on this

### CODE REVIEW

File: src/Stratis.Bitcoin/Base/TipsManager.cs [hunk 1]
```diff
@@ -132,6 +144,16 @@ public ChainedHeader GetLastCommonTip()
...
         }
 
+        /// <inheritdoc />
+        public void CommitTipPersisted(ITipProvider provider, HashHeightPair tip)
+        {
+            ChainedHeader header = this.concurrentChain.GetBlock(tip.Hash);
+            if (tip == null)
+            {
+                throw new ArgumentException(string.Format("Invalid Hash: {0} not found in current chain", tip.Hash));
+            }
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Base/TipsManager.cs]: Aren't you forgetting a call here?

Comment (MithrilMan) [reply src/Stratis.Bitcoin/Base/TipsManager.cs]: yes the other CommitTipPersisted overload

File: src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs [hunk 2]
```diff
@@ -16,7 +17,7 @@ namespace Stratis.Bitcoin.Features.BlockStore
     /// <summary>
     /// <see cref="IBlockRepository"/> is the interface to all the logics interacting with the blocks stored in the database.
     /// </summary>
-    public interface IBlockRepository : IBlockStore
+    public interface IBlockRepository : IBlockStore, ITipProvider
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: Does this really need to be on the interface? just keep it on the implementation no?

Comment (MithrilMan) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: aren't supposed all IBlockRepository implementation to handle the Tip?
If yes than it makes sense to have it at interface level

**Comment by fassadlr** [comment]
@MithrilMan can you update this PR and fix conflicts?

**Comment by MithrilMan** [comment]
this is old and was a WIP that has been interrupted, I think would be better to just close this and start from scratch when time will come

**Comment by fassadlr** [comment]
@MithrilMan closing as we can revisit later 👍

**Pull request closed without merging**